### PR TITLE
Fixing create_cluster.sh 

### PR DESCRIPTION
**Pull request by stewartbutler**

### CODE REVIEW

File: perf/istio-install/create_cluster.sh [hunk 1]
```diff
@@ -141,7 +133,7 @@ function gc() {
 
   # shellcheck disable=SC2236
   set +u
```
Comment (howardjohn) [inline perf/istio-install/create_cluster.sh]: don't need this (and probably the shellcheck disable) if we are doing the `:-`

File: perf/istio-install/create_cluster.sh [hunk 2]
```diff
@@ -154,39 +146,49 @@ function gc() {
...
-  --enable-ip-alias \
-  --metadata disable-legacy-endpoints=true \
-  ${NETWORK_SUBNET} \
-  --default-max-pods-per-node "${MAXPODS_PER_NODE}" \
-  --addons "${ADDONS}" \
-  --enable-network-policy \
-  --workload-metadata-from-node=EXPOSED \
-  --enable-autoupgrade --enable-autorepair \
-  --labels csm=1,test-date=$(date +%Y-%m-%d),version=${ISTIO_VERSION},operator=user_${USER}
+if [[ "$(gcloud beta container --project "${PROJECT_ID}" clusters list --filter=name="${CLUSTER_NAME}" --format='csv[no-heading](name)')" ]]; then
```
Comment (howardjohn) [inline perf/istio-install/create_cluster.sh]: should we use `gc` or gcloud?

File: perf/istio-install/create_cluster.sh [hunk 3]
```diff
@@ -261,18 +264,41 @@ data:
...
+  kubectl create clusterrolebinding cluster-admin-binding \
+    --clusterrole=cluster-admin \
+    --user="$(gcloud config get-value core/account)"
+fi
 
 # Update the cluster with the GCP-specific configmaps
-kubectl -n kube-system apply -f "${CLUSTER_NAME}/configmap-neg.yaml"
-kubectl -n kube-system create secret generic google-cloud-key  --from-file key.json="${CLUSTER_NAME}/google-cloud-key.json"
+if ! kubectl -n kube-system get secret google-cloud-key > /dev/null 2>&1; then 
+  kubectl -n kube-system create secret generic google-cloud-key  --from-file key.json=<(echo "${CLOUDKEY}")
```
Comment (howardjohn) [inline perf/istio-install/create_cluster.sh]: does create fail if it already exists? If it doesn't it seems reasonable to do this always

File: perf/istio-install/create_sa.sh [hunk 4]
```diff
@@ -46,17 +46,31 @@ PROJECT_ID=${PROJECT_ID:?"project id is required"}
 GCP_SA=${1:-istio-data}
 GCP_CTL_SA=${2:-istio-control}
 
-gc iam service-accounts create "${GCP_SA}" # --display-name 'Istio data plane account'
-gc iam service-accounts create "${GCP_CTL_SA}" #--display-name '"Istio control plane account"'
+if [[ -z "$(gcloud beta iam service-accounts --project="${PROJECT_ID}" list --filter=email="${GCP_SA}@${PROJECT_ID}.iam.gserviceaccount.com" --format='csv[no-heading](email)')" ]]; then
+  gc beta iam service-accounts create --quiet "${GCP_SA}" # --display-name 'Istio data plane account'
```
Comment (howardjohn) [inline perf/istio-install/create_sa.sh]: is display-name a real argument?

Comment (stewartbutler) [reply perf/istio-install/create_sa.sh]: It's real, but not required.

File: perf/istio-install/create_cluster.sh (hunk 3 above)
Comment (stewartbutler) [reply perf/istio-install/create_cluster.sh]: Another part of the code conditionally generates the google-cloud-key only if the secret is missing from one of the two namespaces. If we don't have to create another service account key, we shouldn't. I suppose the condition could be changed to check if the CLOUDKEY variable is empty instead, if you would prefer.

File: perf/istio-install/create_cluster.sh (hunk 2 above)
Comment (stewartbutler) [reply perf/istio-install/create_cluster.sh]: The 'gc' function adds some additional flags that aren't compatible with all commands, which includes this one.

File: perf/istio-install/create_cluster.sh (hunk 1 above)
Comment (stewartbutler) [reply perf/istio-install/create_cluster.sh]: Do you want me to submit a PR for this? If not, I'll clean it up next time I touch the file.

File: perf/istio-install/create_sa.sh [hunk 5]
```diff
@@ -46,17 +46,31 @@ PROJECT_ID=${PROJECT_ID:?"project id is required"}
...
+fi
+if [[ -z "$(gcloud beta iam service-accounts --project="${PROJECT_ID}" list --filter=email="${GCP_CTL_SA}@${PROJECT_ID}.iam.gserviceaccount.com" --format='csv[no-heading](email)')" ]]; then
+  gc beta iam service-accounts create --quiet "${GCP_CTL_SA}" #--display-name '"Istio control plane account"'
+fi
+
+function gcloud_get_iam {
+  local sa="${1?"Must pass a SA as first argument to gcloud_get_iam function"}"
+  local role="${2?"Must pass a role as second argument to gcloud_get_iam function"}"
+  gcloud projects get-iam-policy "${PROJECT_ID}" --format=json | jq ".bindings[] | select((.members[] | contains(\"serviceAccount:${sa}@${PROJECT_ID}.iam.gserviceaccount.com\")) and .role == \"roles/${role}\")| .members[0]" -Mr
+
```
Comment (carolynhu) [inline perf/istio-install/create_sa.sh]: When I run this locally, i got this command not found error, here is the log:

```
tools/perf/istio-install/create_sa.sh: line 59: jq: command not found
gcloud projects add-iam-policy-binding --quiet carolyntestprj-262400 --role roles/compute.admin --member serviceAccount:carolyn-test-v4-control@carolyntestprj-262400.iam.gserviceaccount.com
```

Comment (howardjohn) [reply perf/istio-install/create_sa.sh]: https://stedolan.github.io/jq/ install this

**Pull request merged**